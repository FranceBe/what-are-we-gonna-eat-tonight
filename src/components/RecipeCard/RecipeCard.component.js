import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { colors, spaces } from 'styles/variables'
import { mediaMin } from 'styles/mediaMixins'

export const CardContainer = styled.div`
  background-color: ${colors.neutral_lightest};
  border: ${spaces.xtraSmall} solid ${colors.primary_dark};
  display: flex;
  flex-direction: column;
  width: 80%;
  ${mediaMin.md(css`
    width: 90%;
  `)};
`
export const Title = styled.div`
  font-weight: 700;
  text-decoration: underline;
  align-self: center;
  padding: ${spaces.small};
`

export const ImageContainer = styled.div`
  min-width: ${spaces.xtraLarge};
  min-height: ${spaces.xtraLarge};
  background-image: ${props => `url(${props.image})`};
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
`
export const PlaceHolder = styled(ImageContainer)`
  background-color: ${colors.primary_light};
  display: flex;
  align-items: center;
  justify-content: center;
`
export const ImageIngredientsAndLink = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: ${spaces.medium};
`

export const IngredientsAndLink = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${spaces.medium};
`

export class RecipeCard extends React.PureComponent {
  static propTypes = {
    title: PropTypes.string,
    ingredients: PropTypes.string,
    href: PropTypes.string,
    thumbnail: PropTypes.string,
  }

  render() {
    return (
      <CardContainer>
        <Title>{this.props.title}</Title>
        <ImageIngredientsAndLink>
          {this.props.thumbnail
            ? <ImageContainer image={this.props.thumbnail}/>
            : <PlaceHolder> <span>Pas d&apos;image disponible :( </span></PlaceHolder>
          }

          <IngredientsAndLink>
            {this.props.ingredients}
            <a href={this.props.href}> Lien vers la recette </a>
          </IngredientsAndLink>
        </ImageIngredientsAndLink>
      </CardContainer>
    )
  }
}

export default RecipeCard
