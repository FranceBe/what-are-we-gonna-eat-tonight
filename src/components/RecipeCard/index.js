import RecipeCardComponent, { RecipeCard } from 'components/RecipeCard/RecipeCard.component'

export { RecipeCard }

export default RecipeCardComponent
