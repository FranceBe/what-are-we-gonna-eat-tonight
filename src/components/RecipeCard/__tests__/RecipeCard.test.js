import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import RecipeCard,
{ ImageContainer, CardContainer, ImageIngredientsAndLink, IngredientsAndLink, PlaceHolder, Title }
  from 'components/RecipeCard/RecipeCard.component'

describe('RecipeCard', () => {
  describe('style-components', () => {
    describe('ImageContainer', () => {
      it('should match snapshot', () => {
        const ImageContainerTree = renderer.create(<ImageContainer/>).toJSON()
        expect(ImageContainerTree).toMatchSnapshot()
      })
    })
    describe('CardContainer', () => {
      it('should match snapshot', () => {
        const CardContainerTree = renderer.create(<CardContainer/>).toJSON()
        expect(CardContainerTree).toMatchSnapshot()
      })
    })
    describe('ImageIngredientsAndLink', () => {
      it('should match snapshot', () => {
        const ImageIngredientsAndLinkTree = renderer.create(<ImageIngredientsAndLink/>).toJSON()
        expect(ImageIngredientsAndLinkTree).toMatchSnapshot()
      })
    })
    describe('IngredientsAndLink', () => {
      it('should match snapshot', () => {
        const IngredientsAndLinkTree = renderer.create(<IngredientsAndLink/>).toJSON()
        expect(IngredientsAndLinkTree).toMatchSnapshot()
      })
    })
    describe('PlaceHolder', () => {
      it('should match snapshot', () => {
        const PlaceHolderTree = renderer.create(<PlaceHolder/>).toJSON()
        expect(PlaceHolderTree).toMatchSnapshot()
      })
    })
    describe('Title', () => {
      it('should match snapshot', () => {
        const TitleTree = renderer.create(<Title/>).toJSON()
        expect(TitleTree).toMatchSnapshot()
      })
    })
  })
  describe('component rendering', () => {
    const defaultProps = {
      title: 'Chocolate Cake',
      href: 'http://siteweb.site.web',
      ingredients: 'chocolat, cake, chocolate cake',
      thumbnail: 'https://upload.wikimedia.org/wikipedia/commons/d/df/Chocolate_cake.jpg',
    }
    describe('given default props', () => {
      it('should render CardContainer,' +
        'Title, ImageIngredientsAndLink, ImageContainer, IngredientsAndLink with content', () => {
        const RecipeCardWrapper = shallow(<RecipeCard {...defaultProps} />)
        expect(RecipeCardWrapper.find(CardContainer)).toHaveLength(1)
        expect(RecipeCardWrapper.find(Title)).toHaveLength(1)
        expect(RecipeCardWrapper.find(Title).text()).toBe(defaultProps.title)
        expect(RecipeCardWrapper.find(ImageIngredientsAndLink)).toHaveLength(1)
        expect(RecipeCardWrapper.find(ImageContainer)).toHaveLength(1)
        expect(RecipeCardWrapper.find(ImageContainer).props().image).toBe(defaultProps.thumbnail)
        expect(RecipeCardWrapper.find(IngredientsAndLink)).toHaveLength(1)
        expect(RecipeCardWrapper.find(IngredientsAndLink).text()).toBe(defaultProps.ingredients
          + ' Lien vers la recette ')
        expect(RecipeCardWrapper.find('a')).toHaveLength(1)
        expect(RecipeCardWrapper.find('a').props().href).toBe(defaultProps.href)
        expect(RecipeCardWrapper.find('a').text()).toBe(' Lien vers la recette ')
      })
    })
    describe('given there is no thumbnail', () => {
      const props = {
        thumbnail: undefined,
      }
      it('should render the PlaceHolder component instead of ImageContainer', () => {
        const RecipeCardWrapper = shallow(<RecipeCard {...defaultProps} {...props} />)
        expect(RecipeCardWrapper.find(ImageContainer)).toHaveLength(0)
        expect(RecipeCardWrapper.find(PlaceHolder)).toHaveLength(1)
      })
    })
  })
})
