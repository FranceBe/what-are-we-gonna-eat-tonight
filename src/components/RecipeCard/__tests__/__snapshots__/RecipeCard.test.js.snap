// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard style-components CardContainer should match snapshot 1`] = `
.c0 {
  background-color: #FFF;
  border: 1px solid #5F4C44;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 80%;
}

@media (min-width:48em) {
  .c0 {
    width: 90%;
  }
}

<div
  className="c0"
/>
`;

exports[`RecipeCard style-components ImageContainer should match snapshot 1`] = `
.c0 {
  min-width: 200px;
  min-height: 200px;
  background-image: url(undefined);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

<div
  className="c0"
/>
`;

exports[`RecipeCard style-components ImageIngredientsAndLink should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
}

<div
  className="c0"
/>
`;

exports[`RecipeCard style-components IngredientsAndLink should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 10px;
}

<div
  className="c0"
/>
`;

exports[`RecipeCard style-components PlaceHolder should match snapshot 1`] = `
.c0 {
  min-width: 200px;
  min-height: 200px;
  background-image: url(undefined);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-color: #FEFEF1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`RecipeCard style-components Title should match snapshot 1`] = `
.c0 {
  font-weight: 700;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding: 5px;
}

<div
  className="c0"
/>
`;
